### PR TITLE
Senate mechanics

### PR DESCRIPTION
**Pull request by LT-Rascek**
Implementation of Senate Mechanics for 0.1.5

* Players holding Rome/Constantinople manually request Senate Support for their reign instead of a simple event
* Players and/or consuls may elevate senators and dynasties
* The holder of Rome/Constantinople may nominate the next consul
* Costs for these actions are dynamized, based on character culture and traits
* AI behavior is provided so Senate can act somewhat autonomously

### CODE REVIEW

**Comment by LT-Rascek** [comment]
Ready for review and merge.

**Review by loup99 — COMMENTED** [review]
Mostly localisation feedback so far, I will take another look at the rest

File: WtWSMS/localization/english/modifiers/BP_senate_l_english.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿l_english:
+ approved_by_the_senate: "Approved by the Senate"
+ approved_by_the_senate_desc: "This character has been approved by the Roman Senate"
+ rejected_by_the_senate: "Rejected by the Senate"
+ rejected_by_the_senate_desc: "This character has been rejected by the Roman Senate"
+ ignored_the_senate: "Ignored the Senate"
+ ignored_the_senate_desc: "This character has ignored the Roman Senate, boosting his prestige at the cost of his vassal's opinions"
+ stalled_the_senate: "Stalled the Senate"
+ stalled_the_senate_desc: "This character has stalled requesting recognition by the Roman Senate, hampering his prestige and vassal opinions..."
```
[the remark this answers is not in the document]
Comment (LT-Rascek) [reply WtWSMS/localization/english/modifiers/BP_senate_l_english.yml:9]: It's a single modifier; being approved by one senate is good enough.

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 2]
```diff
@@ -20,10 +20,25 @@
...
+ BP_nominate_western_consul_desc: "You may nominate the next Western Consul, if you so wish; doing so provides an opinion and prestige boost to the Consul-designate, but may annoy the rest of the Senate."
+ BP_nominate_eastern_consul: "Nominate an Eastern Consul"
+ BP_nominate_eastern_consul_desc: "You may nominate the next Eastern Consul, if you so wish; doing so provides an opinion and prestige boost to the Consul-designate, but may annoy the rest of the Senate."
+ ## Nominate Senators
+ BP_nominate_western_senator: "Nominate a Western Senator"
+ BP_nominate_western_senator_desc: "You may nominate a new Western Senator, if you so wish; doing so provides a bonus to the Senator-designate, but may annoy the rest of the Senate."
+ BP_nominate_eastern_senator: "Nominate a Eastern Senator"
+ BP_nominate_eastern_senator_desc: "You may nominate a new Eastern Senator, if you so wish; doing so provides a bonus to the Senator-designate, but may annoy the rest of the Senate."
+ ## Elevate Dynasties
+ BP_elevate_dynasty_to_western_senate: "Elevate Dynasty to the Western Senate"
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:35]:
```suggestion
 BP_elevate_dynasty_to_western_senate: "Elevate Dynasty to the Roman Senate"
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 3]
```diff
@@ -20,10 +20,25 @@
...
+ BP_elevate_dynasty_to_western_senate_desc: "As this character is the [dynast|E], you may elevate this character's dynasty to the Western Senate, ensuring all male members of the Dynasty can be part of the Senate; doing so provides a bonuses to the dynasty, but may annoy the rest of the Senate."
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:36]:
```suggestion
 BP_elevate_dynasty_to_western_senate_desc: "As this character is the [dynast|E], you may elevate this character's dynasty to the Roman Senate, ensuring all male members of the Dynasty can be part of the Senate; doing so provides a bonuses to the dynasty, but may annoy the rest of the Senate."
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 4]
```diff
@@ -20,10 +20,25 @@
...
- bureaucratic_revoke_title_interaction: "Revoke Title"
- bureaucratic_revoke_title_interaction_desc: "Take a [title|E] from [recipient.GetShortUINameNoTooltip]\n\n@warning_icon! #weak May incur [tyranny|E] unless you have a cause for revocation#!"
- REVOKE_TITLE_BUREAUCRATIC_PROPOSAL: "I have come to the conclusion that you cannot be entrusted with the dignity and responsibility of the [target.GetName]. I therefore request that you relinquish this title forthwith."
- AI_REFUSAL_IS_DE_JURE_UNDER_BUREAUCRATIC: "[target.GetName|U] belongs to [recipient.GetShortUINamePossessive] [de_jure|E]: $VALUE|=+0$"
- AI_TITLE_IS_REALM_CAPITAL_BUREAUCRATIC: "[landed_title.GetName] is [actor.GetShortUINamePossessive] [de_jure_capital|E]: $VALUE|=+0$"
- cultural_acceptance_loss_revokation_bureuacratic: "[target.GetName] was revoked from a ruler of the [recipient.GetCulture.GetName] culture"
\ No newline at end of file
+ #Senate Interactions
+ ## Nominate Consuls
+ BP_nominate_western_consul: "Nominate a Western Consul"
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:25]:
```suggestion
 BP_nominate_western_consul: "Nominate a Roman Consul"
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 5]
```diff
@@ -20,10 +20,25 @@
...
+ BP_nominate_western_consul_desc: "You may nominate the next Western Consul, if you so wish; doing so provides an opinion and prestige boost to the Consul-designate, but may annoy the rest of the Senate."
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:26]:
```suggestion
 BP_nominate_western_consul_desc: "You may nominate the next Roman Consul, if you so wish; doing so provides an opinion and prestige boost to the Consul-designate, but may annoy the rest of the Senate."
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 6]
```diff
@@ -20,10 +20,25 @@
...
+ BP_nominate_eastern_consul: "Nominate an Eastern Consul"
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:27]:
```suggestion
 BP_nominate_eastern_consul: "Nominate a Constantinopolitan Consul"
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 7]
```diff
@@ -20,10 +20,25 @@
...
+ BP_nominate_eastern_consul_desc: "You may nominate the next Eastern Consul, if you so wish; doing so provides an opinion and prestige boost to the Consul-designate, but may annoy the rest of the Senate."
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:28]:
```suggestion
 BP_nominate_eastern_consul_desc: "You may nominate the next Constantinopolitan Consul, if you so wish; doing so provides an opinion and prestige boost to the Consul-designate, but may annoy the rest of the Senate."
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 8]
```diff
@@ -20,10 +20,25 @@
...
+ ## Nominate Senators
+ BP_nominate_western_senator: "Nominate a Western Senator"
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:30]:
```suggestion
 BP_nominate_western_senator: "Nominate a Roman Senator"
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 9]
```diff
@@ -20,10 +20,25 @@
...
+ BP_nominate_western_senator_desc: "You may nominate a new Western Senator, if you so wish; doing so provides a bonus to the Senator-designate, but may annoy the rest of the Senate."
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:31]:
```suggestion
 BP_nominate_western_senator_desc: "You may nominate a new Roman Senator, if you so wish; doing so provides a bonus to the Senator-designate, but may annoy the rest of the Senate."
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 10]
```diff
@@ -20,10 +20,25 @@
...
+ BP_nominate_eastern_senator: "Nominate a Eastern Senator"
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:32]:
```suggestion
 BP_nominate_eastern_senator: "Nominate a Constantinopolitan Senator"
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 11]
```diff
@@ -20,10 +20,25 @@
...
+ BP_nominate_eastern_senator_desc: "You may nominate a new Eastern Senator, if you so wish; doing so provides a bonus to the Senator-designate, but may annoy the rest of the Senate."
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:33]:
```suggestion
 BP_nominate_eastern_senator_desc: "You may nominate a new Constantinopolitan Senator, if you so wish; doing so provides a bonus to the Senator-designate, but may annoy the rest of the Senate."
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 12]
```diff
@@ -20,10 +20,25 @@
...
+ ## Elevate Dynasties
+ BP_elevate_dynasty_to_western_senate: "Elevate Dynasty to the Western Senate"
+ BP_elevate_dynasty_to_western_senate_desc: "As this character is the [dynast|E], you may elevate this character's dynasty to the Western Senate, ensuring all male members of the Dynasty can be part of the Senate; doing so provides a bonuses to the dynasty, but may annoy the rest of the Senate."
+ BP_elevate_dynasty_to_eastern_senate: "Elevate Dynasty to the Eastern Senate"
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:37]:
```suggestion
 BP_elevate_dynasty_to_eastern_senate: "Elevate Dynasty to the Constantinopolitan Senate"
```

File: WtWSMS/localization/english/BP_interactions_l_english.yml [hunk 13]
```diff
@@ -20,10 +20,25 @@
...
+ BP_elevate_dynasty_to_eastern_senate_desc: "As this character is the [dynast|E], you may elevate this character's dynasty to the Eastern Senate, ensuring all male members of the Dynasty can be part of the Senate; doing so provides a bonuses to the dynasty, but may annoy the rest of the Senate."
```
Comment (loup99) [inline WtWSMS/localization/english/BP_interactions_l_english.yml:38]:
```suggestion
 BP_elevate_dynasty_to_eastern_senate_desc: "As this character is the [dynast|E], you may elevate this character's dynasty to the Constantinopolitan Senate, ensuring all male members of the Dynasty can be part of the Senate; doing so provides a bonuses to the dynasty, but may annoy the rest of the Senate."
```

File: WtWSMS/common/decisions/BP_senate_decisions.txt [hunk 14]
```diff
@@ -0,0 +1,500 @@
...
+		}
+		root = { save_scope_as = the_destroyer }
+		every_player = { trigger_event = { id = BP_roman_senate_events.0002 } }
+	}
+	
+	ai_potential = { always = yes }
+	
+	ai_will_do = {
+		#%25 chance every 5 years => 43% in 10, 58% in 15, 68% in 20 years of doing so
+		base = 25
```
[the remark this answers is not in the document]
Comment (LT-Rascek) [reply WtWSMS/common/decisions/BP_senate_decisions.txt:291]: Done; also tweaked modifier ordering so addition is before multiplication (so a Roman Nostalgia Eranshar ruler won't kill the Senate).

File: ChangeLog.txt [hunk 15]
```diff
@@ -1,6 +1,50 @@
...
+- Created two Senate institutions, with Roman and Constantinopolitan Senates, instead of having them on the map as a title
+   - The institution is interactable through the GUI as a new menu item
+   - Senate Interactions:
+      - Request Recognition:
+         - An event fires several days after taking the Senate Seat, where the Consul requests that you request Senatorial recognition
+         - You can Accept, Ignore the Senate, or Stall; all actions costing prestige and confering modifiers
+         - After that event, players may request through the GUI that the Senate recognize them if they chose to stall instead of making a decision
+      - Rather than a decision, players may disestablish the Senate in the GUI if they are not a Roman-type culture or a Roman government
+   - Senatorial Recognition:
+      - Getting the recognition of the Senate confers a modifier which negates short-regin penalties, at the cost of prestige, but also gains an opinion bump with Senators
```
Comment (loup99) [inline ChangeLog.txt:13]:
```suggestion
      - Getting the recognition of the Senate confers a modifier which negates short-reign penalties, at the cost of prestige, but also gains an opinion bump with Senators
```

**Pull request merged**